cosmoz-image-viewer.js

```diff
@@ -347,52 +355,123 @@
 			}
 
 			if (this.hasWindow) {
-				globals.window.document.title = this._detachedWindowTitle;
-				globals.window.ciw.setImages(this._resolvedImages, this.currentImageIndex);
-				globals.window.focus();
-				return;
+				return this._detachToExistingWindow();
 			}
 
+			globals.windowBeforeUnloadHandler = () => {
+				if (globals.windowOpener !== null) {
+					globals.windowOpener._setIsDetached(false);
+				}
+				globals.windowOpener = null;
+				globals.window = null;
+			};
+
+			return this._detachToNewWindow();
+		},
+
+		_detachToNewWindow() {
 			const
 				w = globals.window = window.open(undefined, this.detachedWindowName, this._detachedWindowFeaturesString),
 				windowTemplate = this.$$('#externalWindow'),
-				windowTemplateClone = windowTemplate.content.cloneNode(true);
+				windowTemplateClone = windowTemplate.content.cloneNode(true),
+				setImages = () => w.ciw.setImages(this._resolvedImages, this.currentImageIndex);
 
 			if (w == null) {
 				// if window.open() is blocked (popup blocked, not emited by native user triggered event)
-				return;
+				return w;
 			}
 
 			if (w.ciw != null) {
 				// browser has been refreshed without closing external window, we will reuse an old window
 				w.removeEventListener('beforeunload', globals.windowBeforeUnloadHandler);
 			}
 
-			globals.windowBeforeUnloadHandler = () => {
```

Comment:
Yes, but as the comment on https://github.com/Neovici/cosmoz-image-viewer/pull/40/files#diff-74b9534db40a1d4427d6f1b815c8b4abR382 (badly?) tries to highlight, we need to drop the old event listener when reusing a window, otherwise the references will be wrong.
And to drop the old one we need a reference to the handler, which gets lost if you overwrite the handler earlier.

Reply:
got it, thx!